NDTensors/test/test_diag.jl

```diff
@@ -56,13 +58,19 @@ using .NDTensorsTestUtils: devices_list, is_supported_eltype
     @test @allowscalar S1 ≈ S2
   end
 end
-@testset "DiagTensor contractions" begin
-  t = tensor(Diag([1.0, 1.0, 1.0]), (3, 3))
-  A = randomTensor(Dense, (3, 3))
+@testset "DiagTensor contractions" for dev in devices_list(copy(ARGS))
+  ## TODO add more GPU tests
+  elt = (dev == NDTensors.mtl ? Float32 : Float64)
+  t = tensor(Diag(elt[1.0, 1.0, 1.0]), (3, 3))
+  A = randomTensor(Dense{elt}, (3, 3))
 
   @test contract(t, (1, -2), t, (-2, 3)) == t
   @test contract(A, (1, -2), t, (-2, 3)) == A
   @test contract(A, (-2, 1), t, (-2, 3)) == transpose(A)
+
+  ## Testing sparse contractions on GPU
+  t = tensor(Diag(one(elt)), (3, 3))
+  @test contract(t, (-1, -2), dev(A), (-1, -2))[] == dot(t, A)
```

Comment:
One of the GPU tests randomly failed, I think this test is too strict, best to add a tolerance bound such as `rtol = sqrt(eps(elt))`.